### ISSUE TITLE
No nullability warning for type inference with lambda argument

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#30480]
Method type inference should report a warning that nullability could not be inferred:
```c#
using System;
using System.Runtime.CompilerServices;
[module: NonNullTypes]
interface I<T> { }
class C
{
    static void F<T>(Func<bool, T> f)
    {
    }
    static void G(I<object> x, I<object?> y)
    {
        F(b => { if (b) return x; else return y; }); // warning: cannot infer nullability
    }
}
```
